pkg/state/manager.go

```diff
@@ -73,200 +83,185 @@ func NewManager(
 	}
 }
 
-// SerializeShipMetadata is used by `ship init` to serialize metadata from ship applications to state file
-func (m *MManager) SerializeShipMetadata(metadata api.ShipAppMetadata, applicationType string) error {
-	debug := level.Debug(log.With(m.Logger, "method", "SerializeShipMetadata"))
+type Update func(VersionedState) (VersionedState, error)
 
-	debug.Log("event", "tryLoadState")
-	current, err := m.TryLoad()
+// applies the provided updater to the current state. Returns the new state and err
+func (m *MManager) StateUpdate(updater Update) (State, error) {
+	m.mut.Lock()
+	defer m.mut.Unlock()
+
+	currentState, err := m.TryLoad()
 	if err != nil {
-		return errors.Wrap(err, "load state")
+		return VersionedState{}, errors.Wrap(err, "tryLoad in safe updater")
 	}
 
-	versionedState := current.Versioned()
-	versionedState.V1.Metadata = &Metadata{
-		ApplicationType: applicationType,
-		ReleaseNotes:    metadata.ReleaseNotes,
-		Version:         metadata.Version,
-		Icon:            metadata.Icon,
-		Name:            metadata.Name,
+	updatedState, err := updater(currentState.Versioned())
+	if err != nil {
+		return VersionedState{}, errors.Wrap(err, "run state update function in safe updater")
```

Comment:
any reason not to return `nil` here?

Reply:
Yeah - it's not `nil`able

Reply:
Oh is `State` not an interface?

Reply:
🤦‍♂ 
I wrote that when it returned VersionedState

Reply:
This compiles for me 

```diff
diff --git a/pkg/state/manager.go b/pkg/state/manager.go
index e2842e5e..adee747a 100644
--- a/pkg/state/manager.go
+++ b/pkg/state/manager.go
@@ -92,7 +92,7 @@ func (m *MManager) StateUpdate(updater Update) (State, error) {
 
        currentState, err := m.TryLoad()
        if err != nil {
-               return VersionedState{}, errors.Wrap(err, "tryLoad in safe updater")
+               return nil, errors.Wrap(err, "tryLoad in safe updater")
        }
 
        updatedState, err := updater(currentState.Versioned())
```

Reply:
Yeah - thus the commit (made at the same time as your comment - 10:54)